webserver/jackson/src/main/java/io/helidon/webserver/jackson/JacksonSupport.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.webserver.jackson;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+import io.helidon.common.http.MediaType;
+import io.helidon.webserver.ContentReaders;
+import io.helidon.webserver.ContentWriters;
+import io.helidon.webserver.Handler;
+import io.helidon.webserver.Routing;
+import io.helidon.webserver.ServerRequest;
+import io.helidon.webserver.ServerResponse;
+import io.helidon.webserver.Service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * A {@link Service} and a {@link Handler} that provides Jackson
+ * support to Helidon.
+ */
+public final class JacksonSupport implements Service, Handler {
+
+    private final BiFunction<? super ServerRequest, ? super ServerResponse, ? extends ObjectMapper> objectMapperProvider;
+
+    /**
+     * Creates a new {@link JacksonSupport}.
+     *
+     * @param objectMapperProvider a {@link BiFunction} that returns
+     * an {@link ObjectMapper} when given a {@link ServerRequest} and
+     * a {@link ServerResponse}; must not be {@code null}
+     *
+     * @exception NullPointerException if {@code objectMapperProvider}
+     * is {@code null}
+     */
+    @SuppressWarnings("checkstyle:linelength")
```

Comment:
This warning should never be suppressed. Please format differently.

Reply:
Done